### PR TITLE
Fix settings tests / refactor

### PR DESCRIPTION
**Pull request by dannycoates**
This is a small patch to get fxa-settings unit tests to pass. :trollface: 

Start in `src/models/AppContext.ts`, it's the root of all the other changes. The rest is all about moving mutations and authClient out of the components.

### CODE REVIEW

File: packages/fxa-settings/src/models/AppContext.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
```
Comment (jaredhirsch) [inline packages/fxa-settings/src/models/AppContext.ts:1]: This is a really nice little layer of abstraction :+1:

**Review by jaredhirsch — APPROVED** [review]
LGTM, great work here. It's good to see the data-management complexity pulled up from the presentational components; makes it easier to reason about the UI behavior or the API behavior without having to keep both in your head at once. :shipit:

**Comment by dannycoates** [comment]
@vbudhram have a look too? :) ... lots of changes

File: packages/fxa-settings/src/components/Page2faReplaceRecoveryCodes/index.tsx [hunk 2]
```diff
@@ -46,38 +33,28 @@ export const Page2faReplaceRecoveryCodes = (_: RouteComponentProps) => {
...
+        )
+      );
     }
-  );
+  }, [account, alertBar, l10n]);
 
   useEffect(() => {
-    session.verified && changeRecoveryCodes.execute();
-  }, [session, changeRecoveryCodes]);
+    session.verified && recoveryCodes.length < 1 && replaceRecoveryCodes();
```
Comment (vbudhram) [inline packages/fxa-settings/src/components/Page2faReplaceRecoveryCodes/index.tsx:52]: 👍🏽

**Review by vbudhram — APPROVED** [review]
@dannycoates Nice work! :shipit:

**Comment by Yogita98** [comment]
Hey @dannycoates 
I just pulled in new commits and rebased my branch and got a notification regarding detection of a secret on [this line](https://github.com/Yogita98/fxa/blob/main/packages/fxa-settings/src/components/PageTwoStepAuthentication/index.test.tsx#L36). Not sure if it was intended so just thought of notifying you.

**Comment by jodyheavener** [comment]
@Yogita98 think that's just mock data

**Comment by Yogita98** [comment]
Okay got it!

**Pull request merged**